### DISCUSSION TITLE
Umami dashboard reloading every 3 seconds

### DISCUSSION THREAD

**Discussion by meirroth** [umami-software/umami#1489] [General]
Running umami on Railway, and been pulling from upstream every update. The dashboard is not usable anymore, it reloads the page every 3 seconds..
Any idea what the cause can be?
I don't see any errors in the console.

**Comment by mikecao** [comment]
I'm not sure what could cause that. If it was a widespread issue I'm sure others would have reported it.

**Comment by thomasthk** [comment]
I also came up with the same issue since upgraded to v1.35. The dashboard refresh itself every3 seconds and I can't notice any errors in the console. No luck even upgraded to v1.36.1.
Running on Vercel with database hosted on Supabase.

**Comment by meirroth** [comment]
@mikecao Are there any troubleshooting steps we can take to figure out what went wrong?

**Comment by mikecao** [comment]
Try using a new browser in incognito mode with any extensions install. A lot times extensions do very unexpected things.

**Comment by meirroth** [comment]
@mikecao I tried in incognito browser with no extensions installed, and the issue remains.

**Comment by thomasthk** [comment]
Same here. Tried incognito mode with different browsers with no extensions. Dashboard still reloading every 3 seconds and unusable. I also tried to redeploy on vercel. Same result.

**Comment by mikecao** [comment]
This is so strange. I'm not seeing that behavior and I can't imagine what could even cause it. Can you give me more details, OS, node version, etc.

**Comment by thomasthk** [comment]
Database hosting: Supabase
Umami hosting: Vercel
Node.js version: 14.x and 16.x
Browsers: Chrome and Safari, tried with Incognito mode with no browser extensions.
When the issue arises? Since upgraded to v1.35 and the current latest v1.36.1
Any errors showing at the console? No

**Comment by meirroth** [comment]
I'm running both umami and the db on Railway.app. 
Node.js version: Not sure where/how to see the version..
When the issue arises? I think since upgraded to v1.35 and the current latest v1.36.1
Any errors showing at the console? No

**Comment by mikecao** [comment]
Can either of you grant me access to your install? I don’t need a password, you can just send me your auth token, mike@umami.is

**Comment by meirroth** [comment]
Sure, what auth token do you need, where do I find it?

**Comment by mikecao** [comment]
Under local storage, there is a key called `umami.auth`. I'll also need the url to your instance.

**Comment by meirroth** [comment]
Ok, I sent you the key.

**Comment by meirroth** [comment]
Let me know if you need any thing else.

**Comment by mikecao** [comment]
@meirroth Thank you. I can log in and confirm the behavior. Still have no idea why it's happening but at least I have a path to troubleshoot.

**Comment by wheeleran** [comment]
I've just updated from 1.33.1 to 1.36.1. same issue, docker-compose container on a Synology NAS. After recreating the container it generates a warning 'using beta Middleware not covered by semver.' EDIT: this seems to be fixed by updating the docker-compose image URL to 'ghcr.io/umami-software/umami:postgresql-latest'. if that's the actual solution then i'm dumb.

**Comment by mikecao** [comment]
Can you see if there is any change in the new version v1.37.0?

**Comment by wheeleran** [comment]
done. a bunch of db migration errors, eg. 42P01 ' relation "account.username_unique" does not exist' and P3018 'prisma migrate deploy' and 42P01 'relation "account.username_unique" does not exist'. there's also a "duplicate key value violates unique constraint "session_session_uuid_key".

also, the click-events panel below the location map ins't loading, with a "something went wrong" error message. i've read the release notes for 1.37.1, but a. i've got this error message on all sites, even those i don't track events on, and b. the release notes don't help me understand what i have to DO to migrate my events? i'm on a docker container on a synology NAS; how do i get into my actual database to manually copy events to the new table from _event_old?

could you look at this again please? could you also help docker-compose dummies like me to learn how to run a rollback to a specific version? i've tried adding the version tag to the compose file but am getting a 'manifest unknown' error. 1.33.1 was the last proper working version for me, and i'd like to hide there and wait in safety till this is resolved.

**Comment by meirroth** [comment]
@mikecao I've just update to v1.37.0 and the issue persists.
Here's the deploy log:
```
yarn run v1.22.19
$ npm-run-all check-db update-tracker start-server
$ node scripts/check-db.js
✓ DATABASE_URL is defined.
✓ Database connection successful.
✓ Database tables found.
Prisma schema loaded from prisma/schema.prisma
Datasource "db": MySQL database "railway" at "containers-us-west-5.railway.app:6390"
2 migrations found in prisma/migrations
Following migration have not yet been applied:
02_add_event_data
To apply migrations in development run yarn prisma migrate dev.
To apply migrations in production run yarn prisma migrate deploy.
Running update...
Prisma schema loaded from prisma/schema.prisma
Datasource "db": MySQL database "railway" at "containers-us-west-5.railway.app:6390"
2 migrations found in prisma/migrations
Applying migration `02_add_event_data`
The following migration have been applied:
migrations/
└─ 02_add_event_data/
└─ migration.sql
 
All migrations have been successfully applied.
✓ Database is up to date.
$ node scripts/update-tracker.js
$ node server.js
Listening on port 3000
```
Then this:
```
filters: {
url: undefined,
referrer: undefined,
os: undefined,
browser: undefined,
device: undefined,
country: undefined
},
params: [ 6, 2022-08-22T22:00:00.000Z, 2022-08-23T21:59:59.999Z ]
}
{
table: 'pageview',
column: null,
filters: {
url: undefined,
referrer: undefined,
os: undefined,
browser: undefined,
device: undefined,
country: undefined
},
params: [ 6, 2022-08-21T22:00:00.001Z, 2022-08-22T22:00:00.000Z ]
}
{
table: 'pageview',
column: null,
filters: {
url: undefined,
referrer: undefined,
os: undefined,
browser: undefined,
device: undefined,
country: undefined
},
params: [ 6, 2022-08-22T22:00:00.000Z, 2022-08-23T21:59:59.999Z ]
}
{
table: 'pageview',
column: null,
filters: {
url: undefined,
os: undefined,
browser: undefined,
device: undefined,
country: undefined
},
params: [ 6, 2022-08-22T22:00:00.000Z, 2022-08-23T21:59:59.999Z ]
}
{
table: 'pageview',
column: null,
filters: {
url: undefined,
referrer: undefined,
os: undefined,
....
```

**Comment by wheeleran** [comment]
Then run the following, if container name is 'umami':

first
`sudo docker exec -it umami sh`
then
`yarn prisma migrate resolve --rolled-back "02_add_event_data"`
then
`yarn prisma migrate deploy`
and
`npm run check-db`

After that, do this, if your db name is 'umami-db':

`sudo docker exec -it umami-db bash`

then
`psql -U umami`
to get into a psql shell

then run the query:

`INSERT INTO event (website_id, session_id, created_at, url, event_name)
SELECT website_id, session_id, created_at, url, event_value
FROM _event_old;`

This is what worked for me, only and entirely with lots of help on this git.

**Comment by meirroth** [comment]
@wheeleran I'm hosting on Railway, so not sure how I can run those commands..

**Comment by meirroth** [comment]
@mikecao Can you please help out?

**Comment by mikecao** [comment]
@meirroth I've tried many things but still don't understand why it's happening. I recommend you roll back to a previous working version and maybe try the next version when it comes out to see if it goes away.

**Comment by meirroth** [comment]
Ok, thanks.

**Comment by meirroth** [comment]
Hey @thomasthk, are you still experiencing this issue?

**Comment by meirroth** [comment]
I've got it to work!! Just spun up a new server running the latest version (v1.37.0), connected to the same old database, and it worked!! I have no idea what was causing this issue but the solution couldn't be simpler, I'm glad I cracked it after hours of trying!

@mikecao Are [these logs](https://user-images.githubusercontent.com/12494197/188153860-f3d2d815-df95-4c7a-a4f0-0d1a3330b194.png) normal?

**Comment by meirroth** [comment]
I've spent the last 4 hours testing different configurations on new deploys (on Railway), and I've discovered that the 3 second reload issue only occurs when I spin up a server with a custom script name set (using `TRACKER_SCRIPT_NAME`).
It worked fine if I ran a server without this!

I believe there is a bug related to `TRACKER_SCRIPT_NAME` that is causing this problem.

To demonstrate this, I have two servers, one with this var set and one without, both of which are connected to the same database.
One of the servers has this problem, while the other does not.

**Comment by thomasthk** [comment]
> Hey @thomasthk, are you still experiencing this issue?

My dashboard won't self reloading anymore after I upgraded to the latest version (v1.37.0).

**Comment by meirroth** [comment]
> > Hey @thomasthk, are you still experiencing this issue?
> 
> My dashboard won't self reloading anymore after I upgraded to the latest version (v1.37.0).

Do you think it's a related issue to the one I described here?

**Comment by thomasthk** [comment]
I can't tell because my deploys are running on Vercel and I also have custom script name set, but the dashboard won't self reloading anymore right now.

**Comment by meirroth** [comment]
> I can't tell because my deploys are running on Vercel and I also have custom script name set, but the dashboard won't self reloading anymore right now.

I suggest you try deploying without the custom script name set, and see if it makes a difference.
If it doesn't help then I think your experiencing a different issue and should probably open a separate issue report.

**Comment by thomasthk** [comment]
OK, I will try and will open a sperate issue report if my case is different than yours. Thanks!

**Comment by nbently** [comment]
@meirroth definitely figured out the root cause, when I removed TRACKER_SCRIPT_NAME from my config everything miraculously worked again.

**Comment by meirroth** [comment]
Hi @mikecao, Please see above comment.
Shouldn't this be an issue instead of a discussion?

**Reply by mikecao** [reply]
This seems to be an issue with Railway so there is nothing we can do on our side.

**Comment by meirroth** [comment]
Got it, thanks.
